File: 389_FindTheDifference.py
class Solution:
    def findTheDifference(self, s: str, t: str) -> str:
        letters_dict = {}
        for l in s:
            letters_dict[l] = 1 if l not in letters_dict else letters_dict[l] + 1
        
        for l in t:
            if l not in letters_dict or letters_dict[l] == 0:
                return l
            letters_dict[l] -= 1

File: cpp/389_FindTheDifference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        if (t.length() < s.length()) {
            string temp = t;
            t = s;
            s = temp;
        }
        
        unordered_map<char, int> letter_map;
        for (char c : s)
            letter_map[c] = letter_map.find(c) == letter_map.end() ? 1 : letter_map[c] + 1;
        
        for (char c : t) {
            if (letter_map.find(c) == letter_map.end() || letter_map[c] == 0)
                return c;
            else
                letter_map[c]--;
        }
        return s[0];
    }
};
